docs/389ds/design/audit-json-logging-design.md

```diff
@@ -0,0 +1,74 @@
+---
+title: "Audit JSON Logging Design"
+---
+
+# Audit JSON Logging
+----------------
+
+Overview
+--------
+
+In order to make the audit logging more consumable by standard parsing tool it will written in a JSON format instead of using a partial LDIF format which not LDAP compliant in its current form.
+
+While the old format, after a bit if data massaging, could be replayed using a LDAP client (e.g. ldapmodify), this is most likely not used by most customers.  Instead, customers are looking for actual audit information that can by consumed by well known log parsers.
+
+JSON Design
+------------
+
+```
+[
+    {
+        date: 20243457458475847Z
+        target_dn: DN,
+        bind_dn: DN,
+        client: IP_ADDRESS,
+        conn_id: ####,
+        op_id: ####,
+        result: ##,
```

Comment:
> I wonder if the changetype is really usefull (it is a bit redundant with the operation parameter field name )

Well I thought it might make parsing "easier" as you wouldn't have to check all the operation keys to see if they are set to determine what the operation is, but I agree it is redundant.   I removed it.

I did some more JSON object cleanup around the modify operation (removed redundant "delete_value" and will just use "value" for all op types).

Reply:
> > I wonder if the changetype is really usefull (it is a bit redundant with the operation parameter field name )
> 
> Well I thought it might make parsing "easier" as you wouldn't have to check all the operation keys to see if they are set to determine what the operation is, but I agree it is redundant. I removed it.
> 

Ok I know I just removed the changetype key, but I'm having second thoughts about it.  I think there are benefits to making it easier to find the op key (especially for a parser writer).  Thoughts?

Reply:
Can you describe a case where having the changetype would really help ?
IMHO most application should use a json decoder library to avoid some nasty surprise with escaped characters
 and such libraries provide way access/check the fields, (and checking if the field exist is always a good idea for robustness point of view)  
BTW in python using  `if 'modify' in parsed_entry_log: ` is as simple as  `if parsed_entry_log['changetype'] == "modify':`